tests/forms_tests/widget_tests/test_radioselect.py

```diff
@@ -70,27 +80,337 @@ def test_nested_choices(self):
+            "num",
+            3,
+            html=html,
+        )
+
+    def test_choices_escaping(self):
+        choices = (("bad", "you & me"), ("good", mark_safe("you &gt; me")))
+        html = """
+        <div>
+          <div>
+            <label><input type="radio" name="escape" value="bad">you & me</label>
+          </div>
+          <div>
+            <label><input type="radio" name="escape" value="good">you &gt; me</label>
+          </div>
+        </div>
+        """
+        self.check_html(
+            self.widget(choices=choices),
+            "escape",
+            None,
+            html=html,
+        )
+
+    def test_choices_unicode(self):
+        html = """
+        <div>
+          <div>
+            <label>
+            <input checked type="radio" name="email"
+              value="\u0160\u0110\u0106\u017d\u0107\u017e\u0161\u0111">
+            \u0160\u0110abc\u0106\u017d\u0107\u017e\u0161\u0111</label>
+          </div>
+          <div>
+            <label>
+            <input type="radio" name="email" value="\u0107\u017e\u0161\u0111">
+            abc\u0107\u017e\u0161\u0111</label>
+          </div>
+        </div>
+        """
+        self.check_html(
+            self.widget(choices=[("ŠĐĆŽćžšđ", "ŠĐabcĆŽćžšđ"), ("ćžšđ", "abcćžšđ")]),
+            "email",
+            "ŠĐĆŽćžšđ",
+            html=html,
+        )
+
+    def test_choices_optgroup(self):
+        """
+        Choices can be nested one level in order to create HTML optgroups
+        """
+        html = """
+        <div>
+          <div>
+            <label><input type="radio" name="nestchoice" value="outer1">Outer 1</label>
+          </div>
+          <div>
+            <label>Group &quot;1&quot;</label>
+            <div>
+              <label>
+              <input type="radio" name="nestchoice" value="inner1">Inner 1</label>
+            </div>
+            <div>
+              <label>
+              <input type="radio" name="nestchoice" value="inner2">Inner 2</label>
+            </div>
+          </div>
+        </div>
+        """
+        self.check_html(
+            self.nested_widget,
+            "nestchoice",
+            None,
+            html=html,
+        )
+
+    def test_choices_select_outer(self):
+        html = """
+        <div>
+          <div>
+            <label>
+            <input checked type="radio" name="nestchoice" value="outer1">Outer 1</label>
+          </div>
+          <div>
+            <label>Group &quot;1&quot;</label>
+            <div>
+              <label>
+              <input type="radio" name="nestchoice" value="inner1">Inner 1</label>
+            </div>
+            <div>
+              <label>
+              <input type="radio" name="nestchoice" value="inner2">Inner 2</label>
+            </div>
+          </div>
+        </div>
+        """
+        self.check_html(
+            self.nested_widget,
+            "nestchoice",
+            "outer1",
+            html=html,
+        )
```

Comment:
This (and few others) can be single-lined.

Reply:
Sure, will do. I was assuming black would take care of this... :thinking: